statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,569 @@
+        elif self.se_method=='boot':
+            return self._bootstrap_weights()
+        else:
+            return self._mean_bootstrap_weight(n_rep, bsn)
+
+    def _jackknife_rep_weights(self, c):
+        # get stratum that the cluster belongs in
+        s = self.sfclust[c]
+        nh = self.ncs[s]
+        w = self.weights.copy()
+        # all weights within the stratum are modified
+        w[self.strat == s] *= nh / float(nh - 1)
+        # but if you're within the cluster to be removed, set as 0
+        w[self.sclust == c] = 0
+        return w
+
+    def _bootstrap_weights(self):
+        w = self.weights.copy()
+        clust_count = np.zeros(self.nclust)
+        for s in range(self.nstrat):
+            # how to handle strata w/ only one cluster?
+            w[self.strat == s] *= float(self.ncs[s] - 1) \
+                                         / self.ncs[s]
+            # If there is only one or two clusters then weights wont change
+            if (self.ncs[s] == 1 or self.ncs[s] == 2):
+                continue
+            # array of clusters to resample from
+            ii = np.flatnonzero(self.sfclust == s)
+            # resample them
+            ii_resample = np.random.choice(ii, size=(self.ncs[s]-1))
+            # accumulate number of times cluster i was resampled
+            clust_count += np.bincount(ii_resample,
+                               minlength=max(self.sclust)+1)
+
+        w *= clust_count[self.sclust]
+        return w
+
+    def _mean_bootstrap_weight(self, replicates, bsn):
+        clust_count = np.zeros(self.design.nclust)
+        for r in range(n_rep):
+            for s in range(self.nstrat):
+                w[self.strat == s] *= ((float(self.ncs[s] - 1) \
+                                            / self.ncs[s])**(1/replicates))
+                # If there is only one or two clusters then weights wont change
+                if (self.ncs[s] == 1 or self.ncs[s] == 2):
+                    continue
+                # array of clusters to resample from
+                ii = np.flatnonzero(self.sfclust == s)
+                # resample them
+                ii_resample = np.random.choice(ii, size=(self.ncs[s]-1))
+                # accumulate number of times cluster i was resampled
+                clust_count += np.bincount(ii_resample,
+                                   minlength=max(self.sclust)+1)
+        # avg number of times cluster i was resampled
+        clust_count *= (bsn / replicates)
+        # augment weights
+        w *= clust_count[self.sclust]
+        return w
+
+
+class SurveyStat(SurveyDesign):
+    """
+    Estimation and inference for summary statistics in complex surveys.
+
+    Parameters
+    -------
+    design : SurveyDesign object
+
+    Attributes
+    ----------
+    est : ndarray
+        The point estimates of the statistic, calculated on the columns
+        of data.
+    vc : ndarray
+        The variance-covariance of the estimates.
+    pseudo : ndarray
```

Comment:
Probably better to have `pseudo` be privatized as `_pseudo`, then obtained via a getter method `pseudo(self)`.  We don't generally need getter methods for attributes, but in this case pseudo only makes sense for jackknife.  Also, it would be good if we could compute it on-demand rather than up-front since most people will not need it.